cmake/AdapthipSYCL.cmake

```diff
@@ -25,6 +17,10 @@ function(add_sycl_executable_implementation)
     add_library(${object_lib_name} OBJECT ${test_cases_list})
     add_executable(${exe_name} $<TARGET_OBJECTS:${object_lib_name}>)
 
+    # hipSYCL needs the macro to be called on the executable target. Setting
+    # properties on source files is insufficient.
+    add_sycl_to_target(TARGET ${exe_name} SOURCES ${test_cases_list})
```

Comment:
I think we also need to `add_sycl_to_target()` on the target `object_lib_name`. It seems that the executable is only involved when everything is linked together, while the compilation target is `object_lib_name`. Not setting this causes the system C++ compiler to be used to compile CTS tests, instead of hipSYCL, which would only be invoked for linking. Clearly this is not what we want ;)

Reply:
I fixed which compiler/linker gets invoked. This actually surfaced the fact that the `test_all` target did not use the compiler launcher previously, or did so in an underhanded way. It is a linkage of object files containing SYCL code, so `add_sycl_to_target` should be in order.